### ISSUE TITLE
LLM03: Establish a clear definition of "model contamination" before the term gets away from us

### ISSUE THREAD

**Issue by talesh** [OWASP/www-project-top-10-for-large-language-model-applications#127]
**OVERVIEW:**
This is a proposal to add a clear definition of the term "model contamination" to v1.1 of LLM03 because it is loosely defined in the community right now and not having a clear definition of the term will muddy future discussion on the safety of LLMs

**PROBLEM:**
The term "contamination" has become a contentious topic within community discussions about LLM security. The ambiguity surrounding its definition has led to inconsistent actions, including the deranking of open-source models by multiple groups. The absence of a standardized definition for "contamination" has resulted in fragmented approaches to evaluating and addressing potential security threats in LLMs. Different groups are using varied criteria for what constitutes "contamination," leading to inconsistent evaluations and potentially unwarranted actions against specific models and datasets.

**GOAL:**
Establish a clear and universally accepted definition of "contamination" in the context of data training poisoning, promoting consistent evaluation criteria and ensuring the protection and integrity of LLMs. We need to distinguish "model contamination" from "training data poisoning", acknowledging the varying degrees of model insecurity can happen due to wilful or poor selection of training data. We need to define "model contamination" before the term gets away from use and should be context-driven and not merely based on subjective disagreements or preferences regarding training content.

**REFERENCES:**
[Contaminated models have been removed from HuggingFace leaderboard](https://twitter.com/FZaslavskiy/status/1692936392509104398)
[OpenPlatypus claims they've cleaned up contaminated HuggingFace datasets](https://twitter.com/natanielruizg/status/1690048217415618560?s=20)

**Comment by guerilla7** [comment]
Hello @talesh 

Just to make sure we are on the same page how we understand Model Contamination:

["Data contamination, i.e., the presence of test data from downstream tasks in the training data of large language models (LLMs)"](https://arxiv.org/pdf/2308.08493.pdf)

Which basically means the test examples were included in the model’s training process (splitting test/training) - when training examples find their way into the test set, the dataset is said to be contaminated. A contaminated test set will provide misleading results because it will evaluate the model on examples it has already seen.

Model contamination is more of a training data, process, and overall quality issue versus a security issue.  In the context of security, poisoned training data or training data poisoning clearly suggest malicious intent to control the output of the ML model.  

Imho, in the context of security, we don't need to redefine model contamination and continue to focus on training data poisoning.  Let me know your thoughts.

Thanks!

**Comment by talesh** [comment]
Hi @guerilla7, 

Thanks for linking that paper, I hadn't seen it before raising this issue in Slack and Github. I'll dive a bit more into it over the weekend.

However in the mean time: I like your proposed distinct definitions of Data Poisoning vs Data Contamination. With those it is clear that there are two patterns that can affect the training data of a model, however only one can be considered a security vulnerability.

The issue that I'm trying to raise is that due to the semantic similarity of both words being used, it might be a good idea to draw that line of distinction so that teams relying on the LLM Top Ten will focus on "poisoning" and not get distracted with "contamination". This will also establish a clear definition of "contamination" in the community so discussions around it are clear going forward.

I am not sure what the strategy around defining exclusionary terms like "contamination" in a TopTen but I believe it would very useful to have in v1.1 of LLM03 for ensuring clarity of definitions.

**Comment by talesh** [comment]
Sounds like a great place for it, then the actual definition doesn't need to be in the TopTen and but rather a reference.

**Comment by skwyddie** [comment]
I think there may be a distinction between a contaminated model and contaminated data, but changing the terminology for the sake of clarity is a good idea. 

For example, is it possible to have training data that contains copyrighted or otherwise undesirable data.  You might also have a model that used bad data or training and is "contaminated", or contains malware within the model file itself. 

From a security perspective, here are two considerations I think should be addressed... somewhere:

1) How do I know that a model was trained with the data set that the model publisher claims it has been trained with (especially challenging when the training process is not owned by/transparent to the adopter of the model)? e.g. how do I know the data set has not been changed since the model was trained? Use case: I download a pre-trained model and dataset from hugging-face. I add adversarial data to the dataset which I use to re-train a backdoor. I publish the model as an All New Thing (tm) and claim to have used the original dataset for training. Obviously, I don't publish the poisoned dataset I actually used. Now, anyone who adopts my model has, in theory, the same functionality as the original published model plus my backdoor.

2) How do I know that the data set that a model was trained with does not contain other problematic data? Problematic data could be copyrighted or sensitive data, toxic content, content facilitating a backdoor, or content that generates other intentional or unintentional bias or fairness concerns upon serving. Use case: I decide to train my own model. Rules suggest that I am only allowed to train models according to the lowest access level of user who is allowed to access the model. So, I have to pare out all of the data classified as "Confidential" from my dataset. I do so and train my model. Later, I learn that the data I used had data in it that has subsequently been reclassified as confidential. Now my model is "spoiled" for its intended use case. Obviously, you can substitute the unintentional inclusion of copyrighted, toxic, or other bias-generating content for this example.

**Comment by talesh** [comment]
Just a quick note @skwyddie / @GangGreenTemperTatum, when this issue was closed, I updated the [Definitions Wiki page](https://github.com/OWASP/www-project-top-10-for-large-language-model-applications/wiki/Definitions) and thought that was enough to close the door on the original issue I raised, but I think the lineage integrity question is worthy of it's own separate discussion.
